microservice/src/test/java/de/gsi/microservice/concepts/majordomo/legacy/CmwLightClient.java

```diff
@@ -0,0 +1,328 @@
+
+    // Field names for the Request Header
+    public static final String EVENT_TYPE_TAG = "eventType";
+    public static final String MESSAGE_TAG = "message";
+    public static final String ID_TAG = "0";
+    public static final String DEVICE_NAME_TAG = "1";
+    public static final String REQ_TYPE_TAG = "2";
+    public static final String OPTIONS_TAG = "3";
+    public static final String CYCLE_NAME_TAG = "4";
+    public static final String ACQ_STAMP_TAG = "5";
+    public static final String CYCLE_STAMP_TAG = "6";
+    public static final String UPDATE_TYPE_TAG = "7";
+    public static final String SELECTOR_TAG = "8";
+    public static final String CLIENT_INFO_TAG = "9";
+    public static final String NOTIFICATION_ID_TAG = "a";
+    public static final String SOURCE_ID_TAG = "b";
+    public static final String FILTERS_TAG = "c";
+    public static final String DATA_TAG = "x";
+    public static final String SESSION_ID_TAG = "d";
+    public static final String SESSION_BODY_TAG = "e";
+    public static final String PROPERTY_NAME_TAG = "f";
+
+    // request type used in request header REQ_TYPE_TAG
+    public static final byte RT_GET = 0;
+    public static final byte RT_SET = 1;
+    public static final byte RT_CONNECT = 2;
+    public static final byte RT_REPLY = 3;
+    public static final byte RT_EXCEPTION = 4;
+    public static final byte RT_SUBSCRIBE = 5;
+    public static final byte RT_UNSUBSCRIBE = 6;
+    public static final byte RT_NOTIFICATION_DATA = 7;
+    public static final byte RT_NOTIFICATION_EXC = 8;
+    public static final byte RT_SUBSCRIBE_EXCEPTION = 9;
+    public static final byte RT_EVENT = 10; // Also used as close
+    public static final byte RT_SESSION_CONFIRM = 11;
+
+    // UpdateType
+    public static final byte UT_NORMAL = (byte) 0;
+    public static final byte UT_FIRST_UPDATE = (byte) 1; // Initial update sent when the subscription is created.
+    public static final byte UT_IMMEDIATE_UPDATE = (byte) 2; //Update sent after the value has been modified by a set call.
+
+    public CmwLightClient() {
+        System.out.println("create new context");
+        controlChannel = context.createSocket(SocketType.DEALER);
+        System.out.println("setup socket");
+        controlChannel.setSndHWM(0);
+        controlChannel.setRcvHWM(0);
+        controlChannel.setIdentity(getIdentity().getBytes()); // hostname/process/id/channel
+        controlChannel.setLinger(0);
+    }
+
+    private String getIdentity() {
+        String hostname;
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            hostname = "localhost";
+        }
+        final long processId = ProcessHandle.current().pid();
+        final long connectionId = 1L;
+        final long channelId = 1L;
+        return hostname + '/' + processId + '/' + connectionId + '/' + channelId;
+    }
+
+    private void receiveData() {
```

Comment:
Codacy found an issue: [The method 'receiveData()' has an NPath complexity of 448, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)